react-front-end/tsrc/drm/DrmHelper.tsx

```diff
@@ -66,8 +71,40 @@ export const createDrmDialog = (
         open={!!drmProtectedHandler}
       />
     );
+  } else if (!isAuthorised) {
+    const violation = await listDrmViolationsInTask(uuid, version);
+    return (
+      <MessageDialog
+        open
+        title="DRM violations"
+        messages={[violation]}
+        close={closeDrmDialog}
+      />
+    );
   } else {
     drmProtectedHandler();
     return;
   }
 };
+
+// Call the API to list DRM violations in a TaskEither where Left is why fail to
+// list violations and Right is the retrieved violation.
+const listDrmViolationsInTask = async (
+  uuid: string,
+  version: number
+): Promise<string> => {
+  const violation = await pipe(
+    TE.tryCatch<Error, OEQ.Drm.DrmViolation>(
+      () => listDrmViolations(uuid, version),
+      (error) => new Error(`${error}`)
+    ),
+    TE.map<OEQ.Drm.DrmViolation, string>(({ violation }) => violation)
+  )();
+
+  return pipe(
+    violation,
+    E.getOrElse(
+      (error) => `Failed to list DRM violations due to ${error.message}`
+    )
+  );
+};
```

Comment:
Does this need to be broken up, or can it just be:

```suggestion
): Promise<string> =>
  await pipe(
    TE.tryCatch<Error, OEQ.Drm.DrmViolation>(
      () => listDrmViolations(uuid, version),
      (error) => new Error(`${error}`)
    ),
    TE.map<OEQ.Drm.DrmViolation, string>(({ violation }) => violation)
    TE.getOrElse(
      (error) => `Failed to list DRM violations due to ${error.message}`
    )
  )();
```

Are are the issues with the async/await?